Content.Server/ReagentOnItem/EntitySystems/ReagentOnItemSystem.cs

```diff
@@ -0,0 +1,91 @@
+using Content.Shared.Chemistry.Components;
+using Content.Shared.Item;
+using Content.Shared.Popups;
+using Content.Shared.ReagentOnItem;
+using Content.Server.Fluids.EntitySystems;
+using Content.Shared.Clothing.Components;
+using Content.Shared.IdentityManagement;
+
+namespace Content.Server.ReagentOnItem;
+
+/// <summary>
+///     The system that helps deal with a genaric reagent being applied to items.
+/// </summary>
+public sealed class ReagentOnItemSystem : EntitySystem
+{
+    [Dependency] private readonly PuddleSystem _puddle = default!;
+    [Dependency] private readonly SharedPopupSystem _popup = default!;
+    public override void Initialize()
+    {
+        base.Initialize();
+    }
+
+    /// <summary>
+    ///     The function tries to apply the given solution mixture to the item.
+    ///     It will apply all reagents that can stick and will spill anything
+    ///     that cannot stick to the floor.
+    /// </summary>
+    /// <returns> Returns false if there is no solution or if entity isn't an item and true otherwise. </returns>
+    public bool AddReagentToItem(EntityUid item, Solution reagentMixture)
+    {
+        if (reagentMixture.Volume <= 0 || !HasComp<ItemComponent>(item))
+        {
+            return false;
+        }
+
+        // Make sure the item doesn't have a non stick surface (E.g jani gloves or boots)
+        if (!HasComp<NonStickSurfaceComponent>(item))
+        {
+            // Yes this code is sussy but its much better than what was used before.
+            // All suspect code for this system is contained here (I hope).
+
+            // Remove all reagents that can actually stick to things and apply them
+            // to the item.
+            var volSpaceLube = reagentMixture.RemoveReagent("SpaceLube", reagentMixture.Volume).Double();
+            var volSpaceGlue = reagentMixture.RemoveReagent("SpaceGlue", reagentMixture.Volume).Double();
+
+            if (volSpaceLube > 0)
+            {
+                var lubed = EnsureComp<SpaceLubeOnItemComponent>(item);
+                AddAndDetermineExtra(lubed, "SpaceLube", volSpaceLube, reagentMixture);
+            }
+            if (volSpaceGlue > 0)
+            {
+                var glued = EnsureComp<SpaceGlueOnItemComponent>(item);
+                AddAndDetermineExtra(glued, "SpaceGlue", volSpaceGlue, reagentMixture);
+            }
```

Comment:
These can be replaced by `TryGetReagent` and should use FixedPoint2 instead of converting to doubles.

Reply:
I converted it to Fixed point!

Would this be the correct way of using TryGetReagent:

```
ReagentId lubename = new ReagentId("SpaceLube", null);

reagentMixture.TryGetReagent(lubename, out var quantity);
```
I tried to look at TryGetReagent but It also appears that it doesn't remove the reagent from the original solution. Would I then have to call RemoveReagent to actually get rid of it after?

Reply:
Methods starting with TryGet all return a bool. So most of the time you'd use the method in an if statement.

In this case you can replace the `reagent > 0` checks with the TryGet.

Reply:
Ahh I see! I think I fixed it.